Proposal for KEDA to become a CNCF Graduated project

tomkerkhove opened the pull request:
Over the past couple of years, we've had the pleasure of seeing KEDA grow from an open-source project that end-users started to play around with to an industry-leading app autoscaler.

Today, we are excited to open our proposal for KEDA to transition to a CNCF Graduated project and are happy to work with the CNCF TOC, TAGs, and community to evaluate our proposal.

Relates to https://github.com/kedacore/governance/issues/63

tomkerkhove commented:
@amye Is there anything we need to do or are we waiting for the next TOC meeting? I believe the freeze period for KubeCon is over so just checking if the action is pending on our side or not.

amye commented:
> @amye Is there anything we need to do or are we waiting for the next TOC meeting? I believe the freeze period for KubeCon is over so just checking if the action is pending on our side or not.

Correct, we have a full TOC schedule for a bit, no action pending on your end.

tomkerkhove commented:
@cathyhongzhang Sure, is there a template that we can use?

cathyhongzhang commented:
Here it is: https://github.com/cncf/toc/blob/main/process/dd-review-template.md. You can also refer to https://github.com/cncf/toc/blob/f41d808146626b7e76bd94d69ffb4ed976f17de3/process/due-diligence-guidelines.md for more info

amye commented on proposals/graduation/keda.md:
```diff
@@ -0,0 +1,186 @@
+
+### * Link to Incubation Due Diligence(DD) Document
+
+The Incubation Due Diligence(DD) document can be found [here](https://docs.google.com/document/d/16iqSoPiOe7N0jNevuXvan31ut3MocGLz-I9yTLGaSdM/edit?usp=sharing).
```
@cathyhongzhang  - this is the incubation due diligence document, this is the starting point for graduation due diligence.
Projects do not need to start over, see SPIFFE/SPIRE for their graduation DD: https://lists.cncf.io/g/cncf-toc/message/7334

tomkerkhove replied:
@cathyhongzhang @amye So I should just extend the existing document or what should I do from my side?

cathyhongzhang replied:
Yes, you can extend and update it. Please refer to the DD of some other projects for reference.

amye commented:
> Here it is: https://github.com/cncf/toc/blob/main/process/dd-review-template.md. You can also refer to https://github.com/cncf/toc/blob/f41d808146626b7e76bd94d69ffb4ed976f17de3/process/due-diligence-guidelines.md for more info

I made a comment in the file pointing out the incubation DD, graduation DD is meant to update that documentation.

cathyhongzhang commented:
I have provided my input/comments to the DD google doc. Please let me know if you have any questions.

tomkerkhove commented:
Thanks, I'll get started on this on Monday

zroubalik commented:
@tomkerkhove wouldn't it be actually better to clone the current DD doc, to keep the old one for a reference?

tomkerkhove commented:
@zroubalik That is what I was planning on doing.

tomkerkhove commented:
While waiting for https://github.com/cncf/toc/issues/980, here is a [copy based on Incubation DD](https://docs.google.com/document/d/1SiGx5xT_z-mXbs88m4FivdxopdZopR4IWbXV9lX1ugM/edit?usp=sharing).

@cathyhongzhang If you DM me your email address I can give you edit rights.

cathyhongzhang commented:
@tomkerkhove sent it to you via slack.

tomkerkhove commented:
Invite setn

cathyhongzhang commented:
Got it. Thanks.

cathyhongzhang commented:
@tomkerkhove  Let's go through the comments and responses in this week's meeting. Please go ahead with setting up the end-user interview. We usually need to talk to at least three end-users adopters, the more the better.

tomkerkhove commented:
For those that want to be interviewed; please fill in [this form](https://forms.gle/91dD4RVxifWsS2hc7).

tomkerkhove commented:
We'd like to schedule a TAG Security review - Do we schedule them with you @lumjjb, @achetal01 & @sublimino? You can find our security review [here](https://github.com/trailofbits/publications/blob/master/reviews/2023-01-keda-securityreview.pdf).

tomkerkhove commented:
> We'd like to schedule a TAG Security review - Do we schedule them with you @lumjjb, @achetal01 & @sublimino? You can find our security review [here](https://github.com/trailofbits/publications/blob/master/reviews/2023-01-keda-securityreview.pdf).

Any thoughts on this @lumjjb, @achetal01 or @sublimino?

xmulligan commented:
I think you need to create an issue in their repo for it https://github.com/cncf/tag-security/issues/1029

tomkerkhove commented:
Thanks, I'll kick that off soon!

@geekygirldawn @jberkus @cathpag Is there a similar doc that explains how to request a review from TAG Contributor Strategy as this seems to be anoter requirement/strong ask

tomkerkhove commented:
Our security audit has now been published on [CNCF blog](https://www.cncf.io/blog/2023/02/02/results-of-the-keda-security-engagement/) and we have tweeted about it [here](https://twitter.com/kedaorg/status/1621507811095244801).

A list of end-users that want to be interviewed have been sent over to @cathyhongzhang as well.

geekygirldawn commented:
We haven't created a specific doc for how to request a review, but here's what you can do:
* Drop a message in the tag-contributor-strategy Slack Channel and highlight any questions / concerns / things you need particular help with
* Drop into one of our meetings to discuss it. You can find our meetings on the CNCF calendar, but the next TAG meeting is next Thursday. If you'd like to have a discussion in the meeting, you can add it to our agenda here: https://docs.google.com/document/d/1Xjw-yAqidQW67zv7OfMRErsfCotc-mfQ_248Te_YL0g/edit#heading=h.e7yodsdpjlas

tomkerkhove commented:
Thanks for getting back to me - The only question is how to do a review as this seems to be required for graduation but it looks like this is not fully in place then. Is that correct summary @geekygirldawn?

zroubalik commented:
Thanks @geekygirldawn , I think we don't have any specific requests, but seems like a review is needed for graduation, so I think that's why Tom asked.

cathyhongzhang commented:
@tomkerkhove where did you send the end-users and their emails? Could you send them to me via slack or my email cathy.h.zhang@intel.com? Thanks.

tomkerkhove commented:
> @tomkerkhove where did you send the end-users and their emails? Could you send them to me via slack or my email cathy.h.zhang@intel.com? Thanks. 

That is indeed where I mailed them to.

geekygirldawn commented:
I'm doing a quick governance review now and will get feedback from @jberkus on it, so we should have something to you by next week. Personally, I don't see any big red flags (unless Josh finds something I missed), but I expect that we'll have some minor feedback / suggestions.

tomkerkhove commented:
Thank you, @geekygirldawn!

tomkerkhove commented:
As per @TheFoxAtWork in https://github.com/cncf/tag-security/issues/1030:
> @tomkerkhove thank you so much for submitting this issue! Since KEDA has applied for graduation, the project should seek a security audit through the Service Desk, as the final audit will supersede the security review. I would recommend - if the project is interested - request a security pal to assist with a lightweight threat model prior to the audit.

So I guess this is closed and we can proceed as they don't have requirements.

jberkus commented:
@tomkerkhove, all:  Governance looks pretty good, really just a few minor nits.

DRAFT governance review (waiting for Dawn to get back) here: https://docs.google.com/document/d/11Qy7uWginP3gFQVpqnNM4CRp4kQKnEM6FTs2M6pm2o8/edit?usp=sharing

tomkerkhove commented:
Thanks @jberkus, already very insightful! Requested access to be able to add comments.

> For that matter, there should be some links from the rest of governance to the CoC, and **the CoC file should be moved somewhere other than the .github directory (how did it get there?)**.

I've noticed this ☝️ The reason it is in `.github` repo is so that it will be linked in all repos in the org  ([GitHub feature](https://www.freecodecamp.org/news/how-to-use-the-dot-github-repository/)).

![image](https://user-images.githubusercontent.com/4345663/218249721-f0d4d230-6a72-49c4-aada-f63b6d9aac0e.png)

 Is this approach discouraged?

geekygirldawn commented:
The governance review doc looks good to me!

tomkerkhove commented:
Thank you @geekygirldawn & @jberkus - I have opened issues for all the feedback and started working on them. You can track them in https://github.com/kedacore/governance/issues/93.

I have updated our DD doc with the link to report, required items and all relevant links from the KEDA side: https://docs.google.com/document/d/1SiGx5xT_z-mXbs88m4FivdxopdZopR4IWbXV9lX1ugM/edit#heading=h.85oterov9i0t

geekygirldawn commented:
@tomkerkhove Yes, you can consider this to be the final version. If you have any more questions / concerns (now or later down the road), please feel free to reach out!

tomkerkhove commented:
TAG runtime review/presentation is being scheduled in https://github.com/cncf/tag-runtime/issues/53

tomkerkhove commented:
https://blog.palark.com/cncf-cloud-native-projects-usage-stats-2022/

> KEDA, which helps autoscale event-driven Kubernetes workloads, accounts for 11.8% of production users. Compared to just 4.7% a year ago, this also indicates a significant boost in adoption.

cathyhongzhang commented:
@tomkerkhove This is great! Thanks for sharing this!

tomkerkhove commented:
Yesterday, we've had the pleasure to present KEDA's changes since Incubation to TAG-Runtime for which you can find the notes [here](https://docs.google.com/document/d/1k7VNetgbuDNyIs_87GLQRH2W5SLgjgOhB6pDyv89MYk/edit#heading=h.ubi0un4zylo0) and our presentation [here](https://docs.google.com/presentation/d/1A9PkpM7Q6gbzmfHv_YMiUlCizXUnyhKhezDvGiLnfS8/edit?usp=sharing).

Thank you for talking to us @raravena80 & @helayoty!

tomkerkhove commented:
> Thank you @geekygirldawn & @jberkus - I have opened issues for all the feedback and started working on them. You can track them in [kedacore/governance#93](https://github.com/kedacore/governance/issues/93).
> 
> I have updated our DD doc with the link to report, required items and all relevant links from the KEDA side: [docs.google.com/document/d/1SiGx5xT_z-mXbs88m4FivdxopdZopR4IWbXV9lX1ugM/edit#heading=h.85oterov9i0t](https://docs.google.com/document/d/1SiGx5xT_z-mXbs88m4FivdxopdZopR4IWbXV9lX1ugM/edit#heading=h.85oterov9i0t)

Thank you @geekygirldawn & @jberkus for the feedback; we have incorporated most of your feedback; super helpful!

tomkerkhove commented:
KEDA maintainers are happy to share that we have finalized the DD document and is ready for review : https://docs.google.com/document/d/1SiGx5xT_z-mXbs88m4FivdxopdZopR4IWbXV9lX1ugM/edit#

I have incorporated 3 end-user interview from Azure Container Apps, Major Gaming Company (Anoymous) & Major IT company (Anonymous) - Thank you, @cathyhongzhang & @rochaporto, for your work! We have more people willing to be interviewed, if there is a need such as Zapier & Grafana Labs. (I think we can share the unanonymized company names with TOC privately, if required).

@TheFoxAtWork @amye @cathyhongzhang @rochaporto  We are more than happy to join the next TOC meeting on June 6th to discuss KEDA's open proposal if there is room for it in the agenda.

amye commented:
We'll let the TOC sponsors confirm that they're ready to review and then move to public comment, we don't try to do any of that in the TOC meetings. Mailing list is where that happens!

amye commented:
/vote

zroubalik commented:
> The vote will be open for `5months 29days 19h 12m`. It will pass if at least `66%` of the users with binding votes vote `In favor 👍`. Once it's closed, results will be published here as a new comment.

Is this really going to be open for half a year? 😅

amye commented:
> > The vote will be open for `5months 29days 19h 12m`. It will pass if at least `66%` of the users with binding votes vote `In favor 👍`. Once it's closed, results will be published here as a new comment.
> 
> Is this really going to be open for half a year? 😅

No, it'll close when it's passing but we needed a time limit to be open 
cc: @jeefy

uhansen commented:
👍

yaron2 commented:
👍 non-binding

tomkerkhove commented:
/check-vote

tomkerkhove commented:
Just trying this command to see if this is the way to get summary since there is no way to check who gave a :+1:. Is this intended approach @amye? (I won't run this daily, no worries)

berndverst commented:
👍 non-binding

mknet3 commented:
/check-vote

tomkerkhove commented:
/check-vote

amye commented:
/check-vote

Aditya-Narayan-Nayak commented:
> 

👍

xoanmm commented:
/check-vote

JennyLawrance commented:
👍

tomkerkhove commented:
/check-vote

raorugan commented:
Already voted!!  👍

tomkerkhove commented:
Blog post is published on cncf.io

https://www.cncf.io/announcements/2023/08/22/cloud-native-computing-foundation-announces-graduation-of-kubernetes-autoscaler-keda/

@amye I believe we should be able to merge this PR?
